docs/building.md

```diff
@@ -38,10 +38,56 @@ ctest -L performance
    |    TEST\_INTEL\_OPTIMIZATION | OFF      | Option | Test intel optimization and vectorization |
    |    CHECK\_REGISTRY\_PRESSURE | OFF      | Option | Check the registry pressure for Kokkos CUDA kernels |
    |               BUILD\_TESTING | ON       | Option | Multi-testing enablement |
-   | PARTHENON\_DISABLE\_EXAMPLES | OFF      | Option | Toggle building of examples, if regression tests are on, drivers needed by the tests will still be built |   
+   | PARTHENON\_DISABLE\_EXAMPLES | OFF      | Option | Toggle building of examples, if regression tests are on, drivers needed by the tests will still be built |
    |   ENABLE\_INTEGRATION\_TESTS | ${BUILD\_TESTING} | Option | Enable integration tests |
    |    ENABLE\_REGRESSION\_TESTS | ${BUILD\_TESTING} | Option | Enable regression tests |
    |          ENABLE\_UNIT\_TESTS | ${BUILD\_TESTING} | Option | Enable unit tests |
    |               CODE\_COVERAGE | OFF      | Option | Builds with code coverage flags |
+   |       CMAKE\_INSTALL\_PREFIX | machine specific | String | Optional path for shared library installation |
 
 ### NB: CMake options prefixed with *PARTHENON\_* modify behavior.
+
+## Building a shared library
+
+An alternative to building Parthenon alongside a custom app (as in the examples)
+is to first build Parthenon separately as a shared library and then link to it
+in the app.
+
+To build Parthenon as a shared library, provide a *CMAKE\_INSTALL\_PREFIX* path
+to the desired install location to the Parthenon cmake call, then call *make install*:
+
+```bash
+cmake -DCMAKE_INSTALL_PREFIX="$your_install_dir" $parthenon_source_dir
+make -j install
+```
+
+This will also install Kokkos in the same directory; in general, one must be
+able to call Kokkos functions directly when writing performant Parthenon apps.
```

Comment:
I don't recall---do we have the ability to link to an external version of Kokkos? If so, we should try to support that too. Does your Kokkos caching mechanism interfere with that?

Reply:
Here are the relevant lines:
```c++
set(Kokkos_CXX_STANDARD 14)                                                      
if(EXISTS ${Kokkos_ROOT}/CMakeLists.txt)                                         
  add_subdirectory(${Kokkos_ROOT} Kokkos)                                        
elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt)        
  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos Kokkos)           
else()                                                                           
  find_package(Kokkos 3 REQUIRED)                                                
endif()
```
I experimented with a `find_package(Kokkos` call on my own but cmake complained about the lack of a `FindKokkos.cmake` file. It seems like the other options here will compile Kokkos. I thought about fiddling with adding support for linking against Kokkos in this PR but decided that was orthogonal enough to be left to another PR (and maybe parthenon *can* link against Kokkos installs currently and I just don't realize it).

Reply:
I think @AndrewGaspar might be able to give some insight into this, but he is on leave. I have also had trouble with cmake detecting Kokkos. I think they might have had issues with how they were exporting the Kokkos cmake config files, but this can be addressed in another issue as suggested.

Reply:
You need kokkos-3 or higher to get the CMake exported target from kokkos.

Reply:
Yeah, make sure to use Kokkos 3.0 or later. If you're using Spack, make sure it's up to date.

Reply:
Make sure to support external Kokkos.